core/variable_binary_arithmetic.cpp

```diff
@@ -31,7 +31,8 @@ template <class... Ts> struct pair_product {
 template <class... Ts>
 using pair_product_t = typename pair_product<Ts...>::type;
 
-using arithmetic_type_pairs = pair_product_t<float, double, int32_t, int64_t>;
+using arithmetic_type_pairs =
+    pair_product_t<float, double, int32_t, int64_t, bool>;
```

Comment:
So this change implies that we can, e.g, add and multiply `bool`. Is that good?

Reply:
If we follow numpy, then:
- not in-place: `add`, `multiply` and `divide` should work. `subtract` is the only one producing: `numpy boolean subtract, the '-' operator, is deprecated, use the bitwise_xor, the '^' operator, or the logical_xor function instead.`
- in-place: `add` and `multiply`. `subtract` produces the same error and `divide` produces a different error (that I don't think was intended)

I think the reason for that is that it not clear what should happen when you subtract bools. What does -1 equal? `bool a = -1` is `true`, while `bool a{-1}` doesn't compile without an explicit cast.

What makes most sense to be is to have `add` and `multiply` supported, but not `subtract` or `divide`. What do you think?

It should also be consistent, currently only `operator*=` can handle bool on the right side, the other in-place operators can't, while all the non in-place can take bools.

Reply:
Can we just disable these operations between `bool` and `bool`?

Reply:
👍 Now only multiplication between numbers and bool is allowed